4. Stack/priority_tasks/PriorityTasksArnab.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * PriorityTasksArnab.java
+ * Given an array of priority values,we have to check if the value next is of higher priority or not.If yes, then print that value,else
+ * Print -1.At the end consider it as circular array,and for the last value check if any higher value to it exists in the array uptill (n-2)th element as the (n-1)th
+ * element is the last value itself,so we would check before that value only if any value higher to it is present or not.
+ * If yes,then print it,else print -1.
+ * Description:-
+ * Create 2 stacks,1 for index and other for storing values.Now start the loop,if stack is empty then push index and value to respective
+ * stacks,Else check if the if the current value is the maximum value or not,if yes then just put -1 there in output array at
+ * corresponding index.Else continue checking as if a value higher than the current peek value of stack is there or not
+ * If yes then pop from stack and continue popping unless a higher value to the current value os achieved and push them to corresponding
+ * index from index stack.And at the same time check if array index has reached n-1,If yes then break from loop.
+ * Time Complexity-O(n) Space Complexity-O(n)
+ * @author [codebook-2000](https://github.com/codebook-2000)
+ */
+ 
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Stack;
+
+public class PriorityTasksArnab {
+    static int[] solve(int[] arr, int n) {
+        Stack<Integer> st = new Stack<Integer>(); //Creating the stack to store values
+        Stack<Integer> index = new Stack<Integer>();//creating the stack to store respective
+        //indices
+
+        int[] ans = new int[n];//Our output array
```

Comment:
I don't think you are supposed to find the max element in this manner. Please try to think of something better.